File: hardware-mall-backend/src/main/java/com/example/mystore/entity/db/Spu.java
package com.example.mystore.entity.db;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import jakarta.validation.constraints.Size;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName(value = "spu", autoResultMap = true)
public class Spu {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long categoryId;

    @Size(max = 100, message = "商品名称最多 100 字")
    private String name;

    private String subtitle;

    private String description;

    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> images;

    private BigDecimal originalPrice;

    private BigDecimal weight;

    private Integer salesCount;

    private Integer status;

    private Integer isRecommend;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    private Long deleteTime;
}

File: hardware-mall-admin/src/api/admin/spu.ts
import request from '@/utils/request'

export interface Spu {
  id?: number
  categoryId: number
  name: string
  subtitle?: string
  description?: string
  images?: string[]
  originalPrice?: number
  weight?: number
  salesCount?: number
  status?: number
  isRecommend?: number
  createTime?: string
  updateTime?: string
  deleteTime?: number
}

export interface SpuQuery {
  page?: number
  limit?: number
  categoryId?: number
  keyword?: string
}

export const getSpuList = (params?: SpuQuery) => {
  return request.get<any>('/admin/spu/list', { params })
}

export const getSpuById = (id: number) => {
  return request.get<Spu>(`/admin/spu/${id}`)
}

export const createSpu = (data: Spu) => {
  return request.post<Spu>('/admin/spu', data)
}

export const updateSpu = (id: number, data: Spu) => {
  return request.put<Spu>(`/admin/spu/${id}`, data)
}

export const deleteSpu = (id: number) => {
  return request.delete(`/admin/spu/${id}`)
}

export const updateSpuStatus = (id: number, status: number) => {
  return request.put(`/admin/spu/${id}/status`, { status })
}
